File: Lintcode/100-Remove-Duplicates-From-Sorted-Array/RemoveDuplicatesFromSortedArray.java

public class RemoveDuplicatesFromSortedArray {
	/**
     * @param A: a array of integers
     * @return : return an integer
     */
	public int removeDuplicates(int[] nums) {
		if (nums == null) {
			return -1;
		}
		if (nums.length < 2) {
			return nums.length;
		}
		int i = 0;
		int j = 1;
		while (j < nums.length) {
			if (nums[i] != nums[j]) {
				i++;
				nums[i] = nums[j];
			}
			j++;
		}
		return i + 1;
	}
}

File: Leetcode/26-Remove-Duplicates-from-Sorted-Array/RemoveDuplicatesFromSortedArray.py
class Solution(object):
    def removeDuplicates(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        i = 0
        if nums == None or len(nums) <1 :
            return i
        j = i+1
        cur = nums[i]
        length = len(nums)
        while j < length :
            if nums[j] != cur :
                i+=1;
                nums[i] = nums[j]
                cur = nums[i]
            j+=1
        return i+1
        
